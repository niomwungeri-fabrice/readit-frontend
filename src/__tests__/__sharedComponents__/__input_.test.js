import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import TextInput from "../../components/common/Inputs/BasicInput";

const props = {
  type: "text",
  name: "email",
  placeholder: "Email",
  onChange: jest.fn(),
  onBlur: jest.fn(),
  value: ""
};
describe("Login component", () => {
  let component;
  beforeEach(() => {
    component = shallow(<TextInput {...props} />);
  });
  test("should match the snapshot", () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  test("should have input tag for email with initial state", () => {
    expect(component.props().className).toEqual("input-group mb-3");
  });
});
