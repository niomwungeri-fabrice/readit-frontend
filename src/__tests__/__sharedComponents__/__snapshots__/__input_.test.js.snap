// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should match the snapshot 1`] = `
<div
  className="input-group mb-3"
>
  <div
    className="input-group-prepend"
  >
    <span
      className="input-group-text"
      id="basic-addon1"
    >
      <i />
    </span>
  </div>
  <input
    aria-describedby="basic-addon1"
    className="form-control"
    name="email"
    onChange={[MockFunction]}
    placeholder="Email"
    type="text"
    value=""
  />
</div>
`;
